### PR TITLE
feat: [PL-22538]: upgraded go-scm version

### PR DESCRIPTION
**Pull request by mohitg0795**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30958)

### CODE REVIEW

File: 930-delegate-tasks/src/test/java/io/harness/delegate/configuration/InstallUtilsTest.java [hunk 1]
```diff
@@ -64,7 +64,8 @@ public void testGetTerraformConfigInspectDownloadUrlPathWindows() throws Excepti
     boolean useCdn = delegateConfiguration.isUseCdn();
 
     assertThat(useCdn).isFalse();
-    assertThat(InstallUtils.getTerraformConfigInspectDownloadUrl(delegateConfiguration, terraformConfigInspectVersion))
+    assertThat(io.harness.delegate.configuration.InstallUtils.getTerraformConfigInspectDownloadUrl(
```
Comment (harshjain12) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/configuration/InstallUtilsTest.java:67]: why this change ?

**Comment by harshjain12** [comment]
Please remove all non required changes

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30958)* status: 10 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[930-delegate-tasks/src/test/java/io/harness/delegate/configuration/InstallUtilsTest.java, line 67 at r1](https://reviewable.io/reviews/harness/harness-core/30958#-Mwc6DL_nSJ6rUOtrVow-r1-67:-MweQKmMDzDyyaPhiQDC:btjyu) ([raw file](https://github.com/harness/harness-core/blob/9ccc5189e583c76a2cae1e857d3da56e349c7691/930-delegate-tasks/src/test/java/io/harness/delegate/configuration/InstallUtilsTest.java#L67)):*
<details><summary><i>Previously, harshjain12 (Harsh Jain) wrote…</i></summary><blockquote>

why this change ?
</blockquote></details>

Intellij pushed those changes directly, may be triggered by some hook. Removed them now.

**Review by vistaarjuneja — APPROVED** [review]
:lgtm:

> Reviewed 9 of 10 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30958#-:-MweY5cc8pheb-wYq5OY:bnfp4nl)* status: 10 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by shubham149 — APPROVED** [review]
> Reviewed 9 of 10 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30958)* status: 1. unresolved discussions.

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 9 of 10 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30958#-:-MwfHGM9A14By1Q7BVAT:bnfp4nl)* status: :shipit: complete! ✓
